extensions/amp-story/0.1/amp-story-hint.js

```diff
@@ -161,18 +153,29 @@ export class AmpStoryHint {
     this.showHint_(FIRST_PAGE_OVERLAY_CLASS);
   }
 
+  /** @visibleForTesting */
+  fadeOutAfterTimeout() {
+    this.hintTimeout_ =
+        this.timer_.delay(() => this.fadeOut_(), NAVIGATION_OVERLAY_TIMEOUT);
+  }
+
   /**
    * Hide all navigation hints.
    */
   hideAllNavigationHint() {
-    this.resetContainerStyle_();
-    this.fadeoutHints_();
+    this.fadeOut_();
+
+    if (this.hintTimeout_ !== null) {
+      this.timer_.cancel(this.hintTimeout_);
+      this.hintTimeout_ = null;
+    }
   }
 
   /** @private */
-  resetContainerStyle_() {
-    this.hintContainer_.classList.remove(NAVIGATION_OVERLAY_CLASS);
-    this.hintContainer_.classList.remove(FIRST_PAGE_OVERLAY_CLASS);
+  fadeOut_() {
```

Comment:
This hides the hints immediately right? How about just `hide_()`?

Reply:
Sure, `hideInternal_` to discern from the public method.